clang/test/dpct/helper_files_ref/include/memory.hpp

```diff
@@ -1304,6 +1304,8 @@ namespace deprecated {
 template <typename T>
 using usm_host_allocator = detail::deprecated::usm_allocator<T, sycl::usm::alloc::host>;
 
+template <typename T>
+using usm_device_allocator = detail::deprecated::usm_allocator<T, sycl::usm::alloc::shared>;
```

Comment:
it 's not totally correct to tell sycl::usm::alloc::shared is a device allocator.

Reply:
From the description of [thrust::device_malloc_allocator](https://thrust.github.io/doc/classthrust_1_1device__malloc__allocator.html), thrust::device_malloc_allocator uses device memory, so detail::deprecated::usm_allocator inherited from sycl::usm_allocator should be specified with "usm::alloc::device" instead.  While currently sycl::usm_allocator does not support "usm::alloc::device"  according to https://github.com/intel/llvm/blob/3df87e20569ea63d0a74de525b6d19788dd8afca/sycl/include/sycl/usm/usm_allocator.hpp#L39.
I 'm considering to implement a specific  device allocator to map thrust::device_malloc_allocator in help function header files instead.

Reply:
Synced with compiler team, the reason that sycl::usm_allocator does not support "usm::alloc::device" is :
there's a fundamental C++ problem here with a tradeoff. If device allocations is supported, it breaks a lot of C++ - std::allocate_shared for example will not work.